hax/hax/util.py

```diff
@@ -2281,62 +2281,75 @@ def set_motr_processes_status(self, fid, status, bAdd=False):
                       motr_processes_status)
 
     @repeat_if_fails()
-    def process_dynamic_fidk_lock(self) -> bool:
+    def obj_dynamic_fidk_lock(self, objt: ObjT) -> bool:
         # Acquire lock to update last_updated_base_fidk.
         # This will block until lock is acquired.
         # Will break if any other exception than
         # HAConsistencyException occurs.
         try:
-            while not self.kv.kv_put('fidk_update_lock', 'true', cas=0):
+            while not self.kv.kv_put(f'{objt.name}/fidk_update_lock',
+                                     'true', cas=0):
                 sleep(1)
             return True
         except Exception:
             return False
 
     @repeat_if_fails()
-    def process_dynamic_fidk_unlock(self):
+    def obj_dynamic_fidk_unlock(self, objt: ObjT):
         # Release fidk_update_lock.
         # This will block until released.
         try:
             keys: List[KeyDelete] = [
-                KeyDelete(name='fidk_update_lock', recurse=True),
+                KeyDelete(name=f'{objt.name}/fidk_update_lock', recurse=True),
             ]
             while not self.kv.kv_delete_in_transaction(keys):
                 sleep(1)
         except Exception:
             raise RuntimeError('Unreachable')
 
     @repeat_if_fails()
-    def get_process_next_dynamic_fidk_lock(self) -> int:
-        if self.process_dynamic_fidk_lock():
-            fidk = self.kv.kv_get('last_dynamic_fid_key/process',
+    def get_obj_next_dynamic_fidk_lock(self, objt: ObjT) -> int:
+        if self.obj_dynamic_fidk_lock(objt):
```

Comment:
Just a question: Here if this hax process gains this lock and dies before unlocking.. then wont that object dynamic fid lock will be locked forever?

Reply:
Yes, that is possible. Window is small but quite possible, ideally we need to avoid such a hax failure, but we also need a guard against it.
I think following approach can be taken,
1. append locking hax process fid in as lock owner in the data.
2. Let RC check the lock whenever it receives a HA failure and delete the lock.

Let me add that change. Thank you.

Reply:
okay, but won't this lead to multiple hax processes simultaneously updating the data in the critical section?
Or is this section updated by a single hax process only?

Reply:
We are using `cas=0` parameter during `kv_put()`, so only one of the hax processes will succeed.

Reply:
Oh yes! Thanks!